tests/provider-search-test.js

```diff
@@ -156,6 +156,159 @@ describeApplication('ProviderSearch', () => {
+          ));
+        });
+
+        it('displays the providers sorted by provider name', () => {
+          expect(ProviderSearchPage.providerList[0].name).to.equal('Analytics for everyone');
+          expect(ProviderSearchPage.providerList[1].name).to.equal('Health Associations');
+          expect(ProviderSearchPage.providerList[2].name).to.equal('My Health Analytics 2');
+          expect(ProviderSearchPage.providerList[3].name).to.equal('My Health Analytics 10');
+        });
+
+        it('shows the sort filter of name in the search form', () => {
+          expect(ProviderSearchPage.getFilter('sort')).to.equal('name');
+        });
+
+        it('reflects the sort in the URL query params', function () {
+          expect(this.app.history.location.search).to.include('sort=name');
+        });
+
+        describe('then searching for other providers', () => {
+          beforeEach(() => {
+            ProviderSearchPage.search('analytics');
+          });
+
+          it('keeps the sort filter of name in the search form', () => {
+            expect(ProviderSearchPage.getFilter('sort')).to.equal('name');
+          });
+
+          it('displays the providers sorted by provider name', () => {
+            expect(ProviderSearchPage.providerList[0].name).to.equal('Analytics for everyone');
+            expect(ProviderSearchPage.providerList[1].name).to.equal('My Health Analytics 2');
+            expect(ProviderSearchPage.providerList[2].name).to.equal('My Health Analytics 10');
+          });
+
+          it('shows the sort filter of name in the search form', () => {
+            expect(ProviderSearchPage.getFilter('sort')).to.equal('name');
+          });
+
+          describe('then clicking another search type', () => {
+            beforeEach(() => {
+              return convergeOn(() => {
+                expect(ProviderSearchPage.$searchResultsItems).to.have.lengthOf(3);
+              }).then(() => ProviderSearchPage.changeSearchType('packages'));
+            });
+
+            it('does not display any results', () => {
+              expect(ProviderSearchPage.$searchResultsItems).to.have.lengthOf(0);
+            });
+
+            describe('navigating back to providers search', () => {
+              beforeEach(() => {
+                return ProviderSearchPage.changeSearchType('providers');
+              });
+
+              it('keeps the sort filter of name in the search form', () => {
+                expect(ProviderSearchPage.getFilter('sort')).to.equal('name');
+              });
+
+              it('displays the last results', () => {
+                expect(ProviderSearchPage.$searchResultsItems).to.have.lengthOf(3);
+              });
+
+              it('reflects the sort=name in the URL query params', function () {
+                expect(this.app.history.location.search).to.include('sort=name');
+              });
+            });
+          });
+          describe('then visiting the page with an existing sort', () => {
```

Comment:
I think this should be pulled out of this describe bock, and put directly within the `"sorting providers"` describe block. That way there isn't already a sort happening in one of the other `beforeEach` hooks.

Reply:
Thanks for the recommendation! Just made the change.